src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/SaveUserLevyDeclarationCommandHandler.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Esfa.Recruit.Vacancies.Client.Application.Commands;
+using Esfa.Recruit.Vacancies.Client.Domain.Repositories;
+using MediatR;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Esfa.Recruit.Vacancies.Client.Application.CommandHandlers
+{
+    public class SaveUserLevyDeclarationCommandHandler : IRequestHandler<SaveUserLevyDeclarationCommand>
+    {
+        private readonly ILogger<SaveUserLevyDeclarationCommandHandler> _logger;
+        private readonly IUserRepository _repository;
+        public SaveUserLevyDeclarationCommandHandler(
+            ILogger<SaveUserLevyDeclarationCommandHandler> logger,
+            IUserRepository repository)
+        {
+            _logger = logger;
+            _repository = repository;
+        }
+
+        public async Task Handle(SaveUserLevyDeclarationCommand message, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Updating user with id: {{userId}} with Legacy declaration of {message.DeclaringAsLevyEmployer}", message.UserId);
+            
+            var user = await _repository.GetAsync(message.UserId);
+
+            user.DeclaredAsLevyPayer = message.DeclaringAsLevyEmployer;
```

Comment:
I think a user can be attached to multiple employers so should this be a list of employer/bool combinations?

Reply:
it could just be a single cookie containing the employer account and user id couldn't it?
that way it's indicating who has been prompted... if the **current** user and/or employer doesn't match then that should trigger the prompt.